internal/controllers/addon_operator_utils.go

```diff
@@ -40,3 +49,29 @@ func (r *AddonOperatorReconciler) reportAddonOperatorReadinessStatus(
 	addonOperator.Status.LastHeartbeatTime = metav1.Now()
```

Comment:
We should consider adding a de-bounce here, e.g. only update the timestamp if it's older than half the update interval.
We need this now, because `Addon` events will re-queue the `AddonOperator` object and will result in a lot of extra updates of this timestamp.

Edit: I also only just now noticed that we are not updating `.Status.ObservedGeneration` here.

Reply:
> Addon events will re-queue the AddonOperator

I'm not sure if I understand this fully. The Addon reconciler only reads the global pause from the pause manager and is not involved in updating the AddonOperator object or reporting any condition to it. That should not cause the AddonOperator object to requeue, right?

The only time the AddonOperator object gets requeued is when it successfully completes one reconciliation cycle (regardless of the pause), which occurs every  1 minure (the default `RequeueAfter` time). So the readiness is never reported at an interval less than update interval itself.

Please correct me if I may have misunderstood. :)

Reply:
> Edit: I also only just now noticed that we are not updating .Status.ObservedGeneration here.

We are, right? Or is this mising something - 

https://github.com/openshift/addon-operator/blob/main/internal/controllers/addon_operator_utils.go#L38

Reply:
Sorry about the .ObservedGeneration thing, code was folded in Github and I didn't notice.
And don't mind about the watch-thing either - I got something wrong, works as intended.

Although having multiple reconcilers interact like this, might raise some eyebrows - but not in a bad way :D